cmd/zpool/zpool_main.c

```diff
@@ -1107,6 +1107,11 @@ zpool_do_create(int argc, char **argv)
 		}
 	}
 
+	/* Make new pools default to partition=raw */
+	if (add_prop_list_default(zpool_prop_to_name(ZPOOL_PROP_PARTITION),
+	    "raw", &props, B_TRUE))
+		goto errout;
```

Comment:
I'd suggest we leave the default behavior as "legacy" at least initially to minimize any disruption.  The existing partitioning scheme does provide a few benefits which users probably don't realize they're depending on.

* The primary partition (-part1) created for ZFS is automatically aligned on a 1MB boundary.  This alignment can be critical for performance when creating a pool on some hardware RAID products.
* Leaving a small bit of space at the end of the device (-part9) helps avoid problems when mixing any matching drives from different vendors which can differ very slightly in total capacity.
* Potentially other issues related to **autoreplace**.